examples/seal/seal.c

```diff
@@ -0,0 +1,182 @@
+    WOLFTPM2_SESSION tpmSession;
+    TPM2B_AUTH auth;
+    const char* outputFile = "keyblob.bin";
+    char defaultData[] = "My1Pass2Phrase3";
+    char *userData = defaultData;
+
+    if (argc >= 2) {
+        if (XSTRNCMP(argv[1], "-?", 2) == 0 ||
+            XSTRNCMP(argv[1], "-h", 2) == 0 ||
+            XSTRNCMP(argv[1], "--help", 6) == 0) {
+            usage();
+            return 0;
+        }
+        if (argv[1][0] != '-') {
+            outputFile = argv[1];
+        }
+    }
+    if (argc >= 3) {
+        if (argv[2][0] != '-') {
+            userData = argv[2];
+        }
+    }
+    while (argc > 1) {
+        if (XSTRNCMP(argv[argc-1], "-aes", 4) == 0) {
+            paramEncAlg = TPM_ALG_CFB;
+        }
+        if (XSTRNCMP(argv[argc-1], "-xor", 4) == 0) {
+            paramEncAlg = TPM_ALG_XOR;
+        }
+        argc--;
+    }
+
+    XMEMSET(&storage, 0, sizeof(storage));
+    XMEMSET(&newKey, 0, sizeof(newKey));
+    XMEMSET(&tpmSession, 0, sizeof(tpmSession));
+    XMEMSET(&auth, 0, sizeof(auth));
+
+    printf("TPM2.0 Simple Seal example\n");
+    printf("\tKey Blob: %s\n", outputFile);
+    printf("\tUse Parameter Encryption: %s\n", TPM2_GetAlgName(paramEncAlg));
+
+    rc = wolfTPM2_Init(&dev, TPM2_IoCb, userCtx);
+    if (rc != TPM_RC_SUCCESS) {
+        printf("\nwolfTPM2_Init failed\n");
+        goto exit;
+    }
+
+    /* get SRK */
+    rc = getPrimaryStoragekey(&dev, &storage, TPM_ALG_RSA);
+    if (rc != 0) goto exit;
+
+    if (paramEncAlg != TPM_ALG_NULL) {
+        /* Start an authenticated session (salted / unbound) with parameter encryption */
+        rc = wolfTPM2_StartSession(&dev, &tpmSession, &storage, NULL,
+            TPM_SE_HMAC, paramEncAlg);
+        if (rc != 0) goto exit;
+        printf("TPM2_StartAuthSession: sessionHandle 0x%x\n",
+            (word32)tpmSession.handle.hndl);
+
+        /* set session for authorization of the storage key */
+        rc = wolfTPM2_SetAuthSession(&dev, 1, &tpmSession,
+            (TPMA_SESSION_decrypt | TPMA_SESSION_encrypt | TPMA_SESSION_continueSession));
+        if (rc != 0) goto exit;
+
+    }
+
+    wolfTPM2_GetKeyTemplate_KeySeal(&publicTemplate);
+
+    /* set session for authorization key */
+    auth.size = (int)sizeof(gKeyAuth)-1;
+    XMEMCPY(auth.buffer, gKeyAuth, auth.size);
+
+    printf("Sealing the user secret into a new TPM key\n");
+    rc = wolfTPM2_CreateKeySeal(&dev, &newKey, &storage.handle,
+                                &publicTemplate, auth.buffer, auth.size,
+                                (BYTE*)userData, (int)strlen(userData));
+    if (rc != TPM_RC_SUCCESS) {
+        printf("wolfTPM2_CreateKey failed\n");
+        goto exit;
+    }
+    printf("Created new TPM seal key (pub %d, priv %d bytes)\n",
+        newKey.pub.size, newKey.priv.size);
+
+    /* Save key as encrypted blob to the disk */
+#if !defined(WOLFTPM2_NO_WOLFCRYPT) && !defined(NO_FILESYSTEM)
+    rc = writeKeyBlob(outputFile, &newKey);
+#endif
+//#else
```

Comment:
Please cleanup the print of the key blob. I recommend adding `#ifdef DEBUG_WOLFTPM` around it.